skcosmo/preprocessing/flexible_scaler.py

```diff
@@ -112,30 +105,24 @@ def inverse_transform(self, X_tr):
 class KernelFlexibleCenterer(TransformerMixin, BaseEstimator):
     """Kernel centering method, similar to KernelCenterer,
     but with additional parameters, relative to which centering
-    is carried out:
+    is carried out.
 
-
-    :param ref_cmean: an array with means for each column.
-    :type ref_cmean: array of shape (1, n_features)
-
-    :param ref_mean: an average for the whole kernel matrix
-    :type ref_mean: array
     """
 
-    def __init__(self, ref_cmean=None, ref_mean=None):
+    def __init__(self):
         """Initialize KernelFlexibleCenterer."""
         pass
 
-    def fit(self, K=None, y=None, ref_cmean=None, ref_mean=None):
+    def fit(self, K=None, y=None, K_fit_rows=None, K_fit_all=None):
```

Comment:
If I understand correctly, the reason we have `K_fit_rows` and `K_fit_all` as arguments is basically for when we want to do a `fit_transform` on just, e.g., the test-train kernel and have the column and total means already available?

If so, I think I'd be more in favor of omitting these as arguments, since they are for a rather specific use case and could be easily abused to get a "wrong" kernel centering.

Am I missing a potential case where having these arguments is crucial?

Reply:
I think you're right, however I'd argue that, in the case of very large kernels, it may be far easier to load the means rather than the full kernel. Also in the case of large kernels, this may save non-negligible time (not much but some)